persist/fs/commitlog/reader.go

```diff
@@ -405,12 +409,37 @@ func (r *reader) decodeAndHandleMetadata(
 	namespace.IncRef()
 	namespace.AppendAll(decoded.Namespace)
 
+	var (
+		tags        ident.Tags
+		tagBytesLen = len(decoded.Tags)
+	)
+	if tagBytesLen != 0 {
+		tagsBytes := r.checkedBytesPool.Get(len(decoded.Tags))
+		tagsBytes.IncRef()
+		tagsBytes.AppendAll(decoded.Tags)
+		tagsBytes.DecRef()
+		tagDecoder.Reset(tagsBytes)
```

Comment:
You're going to clone the bytes below, perhaps best to keep a `checked.Bytes` around and pass it to the `tagDecoder`?  Will avoid the pooling/locking of tag bytes.

e.g. 
```go
// some cached checked.Bytes somewhere
tagBytes := checked.NewBytes(nil, nil)


tagBytes.Reset(decoded.Tags)
tagDecoder.Reset(tagBytes)

// .. clone the tags from the decoder
```

Reply:
oh duh, thanks